# Derivatives / Fixed Income Quiz Solutions

This notebook solves the 5 questions shown in the screenshots.

Each question includes:

- the key formula,
- a short explanation in markdown,
- Python code to calculate the answer,
- the final rounded answer in the format requested.

## Question 1 — Pricing a forward contract with storage costs

**Problem data**

- Spot price of silver: $S_0 = 206$
- Storage cost: \$1 per ounce per year, **payable quarterly in advance**
- Annual interest rate: $9\%$ compounded quarterly
- Maturity: 9 months $= 3$ quarters

### Step 1: Quarterly values

The quarterly interest rate is

$$
i = \frac{0.09}{4} = 0.0225
$$

The storage cost per quarter is

$$
c = \frac{1}{4} = 0.25
$$

### Step 2: Forward pricing logic

Because storage is **paid in advance**, the carrying-cost cash flows happen at:

- $t=0$
- $t=0.25$
- $t=0.50$

To get the fair forward price, we accumulate every cost to maturity $T=0.75$ years:

$$
F_0
=
206(1+i)^3
+ 0.25(1+i)^3
+ 0.25(1+i)^2
+ 0.25(1+i)
$$

So the forward price is the future value of:

1. buying the silver today, and  
2. paying all storage costs at the times they are due.

In [1]:
S0 = 206
annual_rate = 0.09
i = annual_rate / 4
storage_per_quarter = 1 / 4
n_quarters = 3

F0_q1 = (
    S0 * (1 + i) ** n_quarters
    + storage_per_quarter * (1 + i) ** 3
    + storage_per_quarter * (1 + i) ** 2
    + storage_per_quarter * (1 + i)
)

print(f"Theoretical forward price = {F0_q1:.6f}")
print(f"Rounded to nearest integer = {round(F0_q1)}")

Theoretical forward price = 221.004468
Rounded to nearest integer = 221


### Final answer for Question 1

$$
\boxed{221}
$$

## Question 2 — Pricing a 1-period European call option

**Problem data**

- Risk-free gross return: $R = 1.05$
- Current stock price: $S_0 = 50$
- Up factor: $u = 1.08$
- Down factor: $d = \frac{1}{1.08}$
- Strike: $K = 52$

### Step 1: Possible stock prices at maturity

$$
S_u = S_0 u = 50(1.08)=54
$$

$$
S_d = S_0 d = \frac{50}{1.08}\approx 46.2963
$$

### Step 2: Call payoffs

For a European call,

$$
C_u = \max(S_u-K,0)=\max(54-52,0)=2
$$

$$
C_d = \max(S_d-K,0)=0
$$

### Step 3: Risk-neutral probability

In a 1-period binomial model,

$$
q = \frac{R-d}{u-d}
$$

### Step 4: Price the option

$$
C_0 = \frac{qC_u + (1-q)C_d}{R}
$$

In [2]:
R = 1.05
S0 = 50
u = 1.08
d = 1 / 1.08
K = 52

Su = S0 * u
Sd = S0 * d
Cu = max(Su - K, 0)
Cd = max(Sd - K, 0)

q = (R - d) / (u - d)
C0_q2 = (q * Cu + (1 - q) * Cd) / R

print(f"Up-state stock price = {Su:.6f}")
print(f"Down-state stock price = {Sd:.6f}")
print(f"Risk-neutral probability q = {q:.6f}")
print(f"Call price = {C0_q2:.6f}")
print(f"Rounded to 2 decimals = {C0_q2:.2f}")

Up-state stock price = 54.000000
Down-state stock price = 46.296296
Risk-neutral probability q = 0.805288
Call price = 1.533883
Rounded to 2 decimals = 1.53


### Final answer for Question 2

$$
\boxed{1.53}
$$

## Question 3 — Minimum-variance hedging

**Problem data**

- Grapefruit juice to be sold in 3 months: $Q = 150{,}000$ pounds
- Orange juice spot price: \$1.20 per pound
- Grapefruit juice spot price: \$1.50 per pound
- Standard deviation of orange juice price changes: $\sigma_O = 0.2$
- Standard deviation of grapefruit juice price changes: $\sigma_G = 0.2$
- Correlation: $\rho = 0.7$
- Contract size: 1 pound

### Step 1: Write the random cash flow

Let:

- $S_G(T)$ = grapefruit juice price at time $T$
- $F_O(T)$ = orange juice futures price at time $T$
- $F_O(0)$ = orange juice futures price today
- $h$ = number of orange juice futures contracts held

Then the time-$T$ cash flow is

$$
\Pi_T = Q\,S_G(T) + h\big(F_O(T)-F_O(0)\big)
$$

Since $F_O(0)$ is fixed, the variance comes from the random part only:

$$
\text{Var}(\Pi_T)
=
\text{Var}\big(QS_G(T) + hF_O(T)\big)
$$

### Step 2: Expand the variance

$$
\text{Var}(\Pi_T)
=
Q^2\sigma_G^2 + h^2\sigma_O^2 + 2Qh\,\rho\sigma_G\sigma_O
$$

### Step 3: Minimize with respect to $h$

Differentiate and set equal to zero:

$$
\frac{d}{dh}\text{Var}(\Pi_T)
=
2h\sigma_O^2 + 2Q\rho\sigma_G\sigma_O = 0
$$

So,

$$
h^*
=
-\,Q\,\rho\,\frac{\sigma_G}{\sigma_O}
$$

Substitute the numbers:

$$
h^*
=
-150{,}000 \times 0.7 \times \frac{0.2}{0.2}
=
-105{,}000
$$

### Interpretation

The negative sign means the farmer should **short** orange juice futures, not go long.

So the minimum-variance hedge is:

- **short 105,000 orange juice futures contracts**, or equivalently
- if the variable $h$ is defined as “contracts purchased,” then $h=-105{,}000$.

In [3]:
Q = 150_000
rho = 0.7
sigma_grapefruit = 0.2
sigma_orange = 0.2

h_star_q3 = -Q * rho * (sigma_grapefruit / sigma_orange)

print(f"Optimal h* = {h_star_q3:,.0f}")
print("Interpretation: short 105,000 orange juice futures contracts.")

Optimal h* = -105,000
Interpretation: short 105,000 orange juice futures contracts.


### Final answer for Question 3

$$
\boxed{h^*=-105{,}000}
$$

That means the farmer should **sell / short 105,000 orange juice futures contracts**.

## Question 4 — Paradox of risk-neutral pricing

In the binomial model, the no-arbitrage option price is determined by:

- the current stock price,
- the up and down factors,
- the risk-free rate,

and **not** by the stock's true physical probability $p$.

So if the true up-move probability changes from $0.5$ to $0.7$, but the arbitrage-free pricing inputs of the binomial tree are otherwise unchanged, then the risk-neutral price of the European put does **not** change.

### Final answer for Question 4

$$
\boxed{P_0 \text{ does not change}}
$$

## Question 5 — Replicating strategy for pricing

**Problem data**

- Risk-free rate: $r = 0.01$
- Bank account grows from $D$ to $(1+r)D = 1.01D$
- Current stock price: $S_{0,0} = 10$
- Down-state stock price: $S_{1,0} = 8$
- Up-state stock price: $S_{1,1} = 15$
- European call strike: $K = 12$

### Step 1: Option payoffs at time 1

$$
C_d = \max(8-12,0)=0
$$

$$
C_u = \max(15-12,0)=3
$$

### Step 2: Build the replicating portfolio

Let:

- $\Delta$ = number of shares held
- $B$ = dollars invested in the bank account at time 0

Then the portfolio must satisfy:

$$
8\Delta + 1.01B = 0
$$

$$
15\Delta + 1.01B = 3
$$

Subtract the first equation from the second:

$$
7\Delta = 3
\quad \Rightarrow \quad
\Delta = \frac{3}{7}
$$

Plug back in:

$$
8\left(\frac{3}{7}\right) + 1.01B = 0
$$

$$
1.01B = -\frac{24}{7}
\quad \Rightarrow \quad
B = -\frac{24}{7(1.01)}
$$

### Step 3: Price at time 0

The option price equals the cost of the replicating portfolio:

$$
C_0 = \Delta S_0 + B
$$

In [5]:
r = 0.01
S0 = 10
Sd = 8
Su = 15
K = 12

Cd = max(Sd - K, 0)
Cu = max(Su - K, 0)

Delta = (Cu - Cd) / (Su - Sd)
B = (Cd - Delta * Sd) / (1 + r)
C0_q5 = Delta * S0 + B

print(f"Delta = {Delta:.6f}")
print(f"B = {B:.6f}")
print(f"Call price = {C0_q5:.6f}")
print(f"Rounded to 1 decimal = {C0_q5:.1f}")

Delta = 0.428571
B = -3.394625
Call price = 0.891089
Rounded to 1 decimal = 0.9


### Final answer for Question 5

$$
\boxed{0.9}
$$

## Final answer summary

| Question | Answer |
|---|---:|
| 1 | **221** |
| 2 | **1.53** |
| 3 | **-105,000** (short 105,000 futures) |
| 4 | **$P_0$ does not change** |
| 5 | **0.9** |

---
These are the values implied by the standard no-arbitrage / binomial / minimum-variance-hedging formulas used in class.